### PR TITLE
Authentication and storage API

### PR DESCRIPTION
**Pull request by PratikDhanave**

### CODE REVIEW

**Comment by rehrumesh** [comment]
@PratikDhanave please look at the code quality report of codacy.

**Comment by PratikDhanave** [comment]
Yes Need some time to update.

File: compute/ec2/instance.go [hunk 1]
```diff
@@ -4,69 +4,99 @@ import (
 	"encoding/base64"
 	"encoding/xml"
 	"fmt"
-	"log"
+	"github.com/scorelab/gocloud-v2/auth"
+	//"log"
```
Comment (rehrumesh) [inline compute/ec2/instance.go:8]: remove unwanted lines

File: compute/ec2/instance.go [hunk 2]
```diff
@@ -77,48 +107,49 @@ func (ec2 *EC2) query(params map[string]string, resp interface{}) error {
...
 	req.URL.RawQuery = query.Encode()
 
-	auth := Auth{"dummy", "dummy"}
+	auth := Auth{AccessKey: auth.Config.AWSAccessKeyID, SecretKey: auth.Config.AWSSecretKey}
 
-	SignV2(req, auth)
+	SignatureV2(req, auth)
 
-	fmt.Println(req)
+	//fmt.Println(req)
```
Comment (rehrumesh) [inline compute/ec2/instance.go:117]: remove unwanted comments

File: compute/ec2/sign.go [hunk 3]
```diff
@@ -70,13 +78,16 @@ func clientToken() (string, error) {
...
 func fprintfWrapper(w io.Writer, format string, vals ...interface{}) func() error {
 	return func() error {
 		_, err := fmt.Fprintf(w, format, vals...)
 		return err
 	}
 }
 
+*/
+/*
 func errorCollector(writers ...func() error) error {
```
Comment (rehrumesh) [inline compute/ec2/sign.go:91]: remove this part

File: compute/gce/instance.go [hunk 4]
```diff
@@ -286,3 +300,36 @@ func (gce *GCE) listnode(request interface{}) (resp interface{}, err error) {
...
+
+	Startnoderesp, err := client.Do(Startnoderequest)
+
+	defer Startnoderesp.Body.Close()
+
+	body, err := ioutil.ReadAll(Startnoderesp.Body)
+
+	fmt.Println(string(body))
+
+	//respq := GCEResponse{}
```
Comment (rehrumesh) [inline compute/gce/instance.go:326]: remove unwanted comments.

File: main.go [hunk 5]
```diff
@@ -146,55 +147,308 @@ func main() {
...
-	disk := []map[string]interface{}{
-		{
-			"Boot":             true,
-			"AutoDelete":       false,
-			"DeviceName":       "bokya",
-			"Type":             "PERSISTENT",
-			"Mode":             "READ_WRITE",
-			"InitializeParams": InitializeParams,
+	/*
+		InitializeParams := map[string]string{
```
Comment (rehrumesh) [inline main.go:151]: Remove this block

File: storage/amazonstorage/storage.go [hunk 6]
```diff
@@ -0,0 +1,321 @@
+package amazonstorage
+
+import (
+	"encoding/xml"
+	auth "github.com/scorelab/gocloud-v2/auth"
+	awsauth "github.com/scorelab/gocloud-v2/awsauth"
+	//"log"
```
Comment (rehrumesh) [inline storage/amazonstorage/storage.go:7]: remove unwanted comments

File: storage/googlestorage/googlestorage.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+type Snapshot struct {
+	Name string `json:"name"`
+}
+
+type Attachdisk struct {
+	Source string `json:"source"`
+}
+
+/*
+type Snapshot struct {
```
Comment (rehrumesh) [inline storage/googlestorage/googlestorage.go]: Remove this block

**Review by rehrumesh — CHANGES_REQUESTED** [review]
Please do the necessary changes to the comments

**Comment by rehrumesh** [comment]
@PratikDhanave There are code conflicts and code quality issues. Fix them asap so we can close this PR.

**Review by rehrumesh — CHANGES_REQUESTED** [review]
update with requests

**Pull request closed without merging**